registry/handlers/api_test.go

```diff
@@ -877,10 +877,7 @@ func testManifestAPI(t *testing.T, env *testEnv, args manifestArgs) (*testEnv, m
 		t.Fatalf("unexpected error signing manifest: %v", err)
 	}
 
-	payload, err := signedManifest.Payload()
-	checkErr(t, err, "getting manifest payload")
-
-	dgst, err := digest.FromBytes(payload)
+	dgst, err := digest.FromBytes(signedManifest.Canonical)
```

Comment:
Might be good to have a help function in the `distribution` package that provides the digest of a manifest. I've noticed this pattern all over the place when I was working on this and didn't go through with it, since the manifest model was odd. With this new interface, we can safely define it. It should pretty much always be `manifest.Payload()`.